chall03/pcuadrad.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+
+import sys
+
+def error():
+    print ("usage: " + sys.argv[0] + " <1-9 squared_rows...>")
+    exit(1)
+
+def snail(array):
+    snailArray = []
+    rowBegin = 0
+    rowEnd = len(array) - 1
+    colBegin = 0
+    colEnd = len(array[0]) - 1
+    while rowBegin <= rowEnd and colBegin <= colEnd:
+        for col in range(colBegin, colEnd + 1):
+            snailArray.append(array[rowBegin][col])
+        rowBegin += 1
+        for row in range(rowBegin, rowEnd + 1):
+            snailArray.append(array[row][colEnd])
+        colEnd -= 1
+        for col in range(colEnd, colBegin - 1, -1):
+            snailArray.append(array[rowEnd][col])
+        rowEnd -= 1
+        for row in range(rowEnd, rowBegin - 1, -1):
+            snailArray.append(array[row][colBegin])
+        colBegin += 1
+    print(', '.join(snailArray))
```

Comment:
If you have already sanitised the input, why do you store all the array in memory instead of printing it while you traverse the whole array, saving memory in the process?

Reply:
For me it is cleaner to do a single print at the end of once, but it is true that it is less efficient, I will take it into account thanks.

Reply:
It's cleaner for me too, and in case there was an error external to the program, you wouldn't have printed anything yet and it would be better. But still, I think printing through the process is more performant.